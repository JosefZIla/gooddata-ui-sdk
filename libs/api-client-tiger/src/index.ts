// (C) 2019-2021 GoodData Corporation
import { tigerClientFactory, ITigerClient } from "./client";
import {
    axios as defaultAxios,
    newAxios,
    setAxiosAuthorizationToken,
    setGlobalAuthorizationToken,
} from "./axios";

export { VisualizationObjectModel } from "./gd-tiger-model/VisualizationObjectModel";
export { AnalyticalDashboardObjectModel } from "./gd-tiger-model/AnalyticalDashboardObjectModel";

export {
    isAttributeHeader,
    isObjectIdentifier,
    isResultAttributeHeader,
    isVisualizationObjectsItem,
    isFilterContextData,
    ResultDimensionHeader,
} from "./gd-tiger-model/typeGuards";

export { newAxios, setAxiosAuthorizationToken, setGlobalAuthorizationToken };

export * from "./generated/afm-rest-api/api";
export * from "./generated/metadata-json-api/api";
export * from "./client";

export { jsonApiHeaders, JSON_API_HEADER_VALUE } from "./constants";

export {
    MetadataUtilities,
    MetadataGetEntitiesResult,
    MetadataGetEntitiesFn,
    MetadataGetEntitiesOptions,
    MetadataGetEntitiesParams,
} from "./metadataUtilities";

const defaultTigerClient: ITigerClient = tigerClientFactory(defaultAxios);

export default defaultTigerClient;
