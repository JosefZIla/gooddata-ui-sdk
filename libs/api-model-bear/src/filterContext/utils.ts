// (C) 2019-2023 GoodData Corporation
import omitBy from "lodash/omitBy";
import isUndefined from "lodash/isUndefined";
import { GdcFilterContext } from "./GdcFilterContext";
import isDateFilter = GdcFilterContext.isDateFilter;
import isAttributeFilter = GdcFilterContext.isAttributeFilter;

function sanitizeDateFilter(filter: GdcFilterContext.IDateFilter): GdcFilterContext.IDateFilter {
    const {
        dateFilter: { from, to, type, granularity },
        dateFilter,
    } = filter;
    const sanitizedProperties = omitBy(dateFilter, isUndefined);

    return {
        dateFilter: {
            type,
            granularity,
            ...sanitizedProperties,
            ...(from === undefined ? {} : { from: String(from) }),
            ...(to === undefined ? {} : { to: String(to) }),
        },
    };
}

function sanitizeAttributeFilter(
    filter: GdcFilterContext.IAttributeFilter,
): GdcFilterContext.IAttributeFilter {
    const {
        attributeFilter: { displayForm, negativeSelection, attributeElements, selectionMode },
    } = filter;

    return {
        attributeFilter: {
            displayForm,
            negativeSelection,
            attributeElements,
            selectionMode,
        },
    };
}

/**
 * @public
 */
export function sanitizeFiltersForExport(
    filters: GdcFilterContext.FilterContextItem[],
): GdcFilterContext.FilterContextItem[] {
    return filters.map((filter: GdcFilterContext.FilterContextItem): GdcFilterContext.FilterContextItem => {
        if (isDateFilter(filter)) {
            return sanitizeDateFilter(filter);
        }
        if (isAttributeFilter(filter)) {
            return sanitizeAttributeFilter(filter);
        }
        return filter;
    });
}
