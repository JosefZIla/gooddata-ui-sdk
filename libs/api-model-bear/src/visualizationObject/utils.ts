// (C) 2007-2022 GoodData Corporation
import {
    IVisualizationObjectContent,
    IBucket,
    BucketItem,
    IVisualizationAttributeContent,
    IMeasureContent,
    IVisualizationObjectMeasureDefinition,
    isVisualizationObjectMeasureDefinition,
    VisualizationObjectFilter,
    VisualizationObjectAttributeFilter,
    isVisualizationObjectAttribute,
    isVisualizationObjectMeasure,
    isVisualizationObjectAttributeFilter,
    isVisualizationObjectPositiveAttributeFilter,
} from "./GdcVisualizationObject.js";
import { IObjUriQualifier } from "../base/GdcTypes.js";

function getAttributesInBucket(bucket: IBucket): IVisualizationAttributeContent[] {
    return bucket.items.reduce((list: IVisualizationAttributeContent[], bucketItem: BucketItem) => {
        if (isVisualizationObjectAttribute(bucketItem)) {
            list.push(bucketItem.visualizationAttribute);
        }
        return list;
    }, []);
}

function getAttributes(mdObject: IVisualizationObjectContent): IVisualizationAttributeContent[] {
    const buckets = mdObject.buckets;
    return buckets.reduce((categoriesList: IVisualizationAttributeContent[], bucket: IBucket) => {
        categoriesList.push(...getAttributesInBucket(bucket));
        return categoriesList;
    }, []);
}

function getMeasuresInBucket(bucket: IBucket): IMeasureContent[] {
    return bucket.items.reduce((list: IMeasureContent[], bucketItem: BucketItem) => {
        if (isVisualizationObjectMeasure(bucketItem)) {
            list.push(bucketItem.measure);
        }
        return list;
    }, []);
}

function getDefinition(
    measure: IMeasureContent,
): IVisualizationObjectMeasureDefinition["measureDefinition"] | undefined {
    return isVisualizationObjectMeasureDefinition(measure.definition)
        ? measure.definition.measureDefinition
        : undefined;
}

function getMeasures(mdObject: IVisualizationObjectContent): IMeasureContent[] {
    const buckets = mdObject.buckets;
    return buckets.reduce((measuresList: IMeasureContent[], bucket: IBucket) => {
        measuresList.push(...getMeasuresInBucket(bucket));
        return measuresList;
    }, []);
}

function getMeasureFilters(measure: IMeasureContent): VisualizationObjectFilter[] {
    return getDefinition(measure)?.filters ?? [];
}

function getMeasureAttributeFilters(measure: IMeasureContent): VisualizationObjectAttributeFilter[] {
    return getMeasureFilters(measure).filter(isVisualizationObjectAttributeFilter);
}

function getAttributeFilters(mdObject: IVisualizationObjectContent): VisualizationObjectAttributeFilter[] {
    return getMeasures(mdObject).reduce(
        (filters: VisualizationObjectAttributeFilter[], measure: IMeasureContent) => {
            filters.push(...getMeasureAttributeFilters(measure));
            return filters;
        },
        [],
    );
}

function getAttributeFilterDisplayForm(measureFilter: VisualizationObjectAttributeFilter): string {
    return isVisualizationObjectPositiveAttributeFilter(measureFilter)
        ? (measureFilter.positiveAttributeFilter.displayForm as IObjUriQualifier).uri
        : (measureFilter.negativeAttributeFilter.displayForm as IObjUriQualifier).uri;
}

/**
 * @public
 */
export function getAttributesDisplayForms(mdObject: IVisualizationObjectContent): string[] {
    const attributesDfs = getAttributes(mdObject).map(
        (attribute: IVisualizationAttributeContent) => (attribute.displayForm as IObjUriQualifier).uri,
    );
    const attrMeasureFilters = getAttributeFilters(mdObject);
    const attrMeasureFiltersDfs = attrMeasureFilters.map(getAttributeFilterDisplayForm);
    return [...attrMeasureFiltersDfs, ...attributesDfs];
}
