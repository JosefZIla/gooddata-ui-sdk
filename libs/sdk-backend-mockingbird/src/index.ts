// (C) 2019-2021 GoodData Corporation
/**
 * @packageDocumentation
 */
import { dummyBackend, dummyBackendEmptyData, dummyDataView } from "@gooddata/sdk-backend-base";
export { dummyBackend, dummyBackendEmptyData, dummyDataView };

export { recordedBackend } from "./recordedBackend";

export {
    RecordingIndex,
    ExecutionRecording,
    InsightRecording,
    DisplayFormRecording,
    ScenarioRecording,
    RecordedBackendConfig,
    CatalogRecording,
    VisClassesRecording,
    DashboardRecording,
    RecordedRefType,
    SecuritySettingsUrlValidator,
    SecuritySettingsOrganizationScope,
} from "./recordedBackend/types";

export {
    recordedDataView,
    recordedDataViews,
    recordedInsight,
    DataViewAll,
    dataViewWindow,
    DataViewFirstPage,
    NamedDataView,
} from "./recordedBackend/execution";

export { CompositeBackendPart, compositeBackend } from "./compositeBackend/index";

export {
    LegacyExecutionRecording,
    legacyRecordedBackend,
    legacyRecordedDataView,
    LegacyRecordingIndex,
    LegacyWorkspaceRecordings,
} from "./legacyRecordedBackend";
