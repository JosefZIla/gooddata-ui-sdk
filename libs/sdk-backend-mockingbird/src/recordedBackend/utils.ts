// (C) 2019-2021 GoodData Corporation

/**
 * @internal
 */
export function identifierToRecording(id: string): string {
    return id.replace(/\./g, "_");
}
