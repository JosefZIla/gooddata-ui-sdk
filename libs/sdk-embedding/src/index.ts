// (C) 2019-2021 GoodData Corporation
/**
 * @packageDocumentation
 */
export {
    IObjectMeta,
    IPostMessageContextPayload,
    EmbeddedGdc,
    GdcErrorType,
    GdcEventType,
    GdcMessageEventListener,
    GdcProductName,
    IGdcMessage,
    IGdcMessageEnvelope,
    IGdcMessageEvent,
    IGdcMessageEventListenerConfig,
    CommandFailed,
    CommandFailedData,
    ICommandFailedBody,
    IDrillableItemsCommandBody,
    ISimpleDrillableItemsCommandBody,
    isCommandFailedData,
    getEventType,
} from "./iframe/common";

export { EmbeddedAnalyticalDesigner } from "./iframe/ad";

export { EmbeddedKpiDashboard } from "./iframe/kd";
