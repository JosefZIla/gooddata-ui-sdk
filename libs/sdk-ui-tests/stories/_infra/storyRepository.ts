// (C) 2022 GoodData Corporation
import flatMap from "lodash/flatMap";
import { toId } from "@componentdriven/csf";
import { storiesOf as storiesOfInternal } from "@storybook/react";

import { IBackstopScenarioConfig } from "./backstopScenario";
import { wrapForBackstop } from "./backstopWrapper";

interface ISingleScreenshotConfig {
    screenshot?: IBackstopScenarioConfig | true;
}

function isSingleScreenshotConfig(obj: unknown): obj is ISingleScreenshotConfig {
    return !!(obj as ISingleScreenshotConfig | undefined)?.screenshot;
}

interface IMultipleScreenshotConfig {
    screenshots?: Record<string, IBackstopScenarioConfig>;
}

function isMultipleScreenshotConfig(obj: unknown): obj is IMultipleScreenshotConfig {
    return !!(obj as IMultipleScreenshotConfig | undefined)?.screenshots;
}

type ScreenshotConfig = ISingleScreenshotConfig | IMultipleScreenshotConfig;

type IAdditionalStoryConfig = ScreenshotConfig;

interface IStoryDefinition {
    name: string;
    render: () => JSX.Element;
    config?: IAdditionalStoryConfig;
}

type Kind = string;

class StoryRepository {
    private stories: Record<Kind, IStoryDefinition[]> = {};

    public addStory(
        kind: Kind,
        name: string,
        render: () => JSX.Element,
        config?: IAdditionalStoryConfig,
    ): void {
        if (!this.stories[kind]) {
            this.stories[kind] = [];
        }
        this.stories[kind].push({
            name,
            render,
            config,
        });
    }

    public toBackstopJson(): string {
        const flattened = flatMap(Object.entries(this.stories), ([kind, stories]) => {
            return flatMap(
                stories.filter(
                    (story) =>
                        isSingleScreenshotConfig(story.config) || isMultipleScreenshotConfig(story.config),
                ),
                (story) => {
                    const base = {
                        storyId: toId(kind, story.name),
                        storyKind: kind,
                        storyName: story.name,
                    };

                    if (isSingleScreenshotConfig(story.config)) {
                        return [
                            {
                                ...base,
                                scenarioConfig:
                                    story.config.screenshot === true ? {} : story.config.screenshot,
                            },
                        ];
                    }

                    if (isMultipleScreenshotConfig(story.config)) {
                        return Object.entries(story.config!.screenshots!).map(
                            ([scenarioName, scenarioConfig]) => {
                                return {
                                    ...base,
                                    scenarioName,
                                    scenarioConfig,
                                };
                            },
                        );
                    }

                    return [];
                },
            );
        });

        return JSON.stringify(flattened, null, 4);
    }

    public populateStorybook(): void {
        Object.entries(this.stories).forEach(([kind, stories]) => {
            stories.forEach((story) => {
                const renderFunction =
                    isSingleScreenshotConfig(story.config) || isMultipleScreenshotConfig(story.config)
                        ? wrapForBackstop(story.render)
                        : story.render;

                storiesOfInternal(kind, module).add(story.name, renderFunction);
            });
        });
    }
}

const singleton = new StoryRepository();

export function storiesOf(kind: Kind) {
    return {
        add(name: string, render: () => JSX.Element, config?: IAdditionalStoryConfig) {
            singleton.addStory(kind, name, render, config);
            return this;
        },
    };
}

export function toBackstopJson(): string {
    return singleton.toBackstopJson();
}

export function populateStorybook(): void {
    return singleton.populateStorybook();
}
