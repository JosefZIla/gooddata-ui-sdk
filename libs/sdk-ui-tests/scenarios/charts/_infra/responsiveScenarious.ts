// (C) 2007-2019 GoodData Corporation
import { ScenarioCustomizer, scenariosFor, UnboundVisProps, VisProps } from "../../../src";

export interface IResponsiveSize {
    label?: string;
    width: number;
    height: number;
}

export function responsiveScenarios<T extends VisProps>(
    chart: string,
    groupNames: string[],
    component: React.ComponentType<T>,
    baseProps: UnboundVisProps<T>,
    sizes: Array<IResponsiveSize>,
    customizer?: ScenarioCustomizer<T>,
) {
    return sizes.map((size) => {
        const groupLabel = size.label ? size.label : `${size.width}x${size.height}`;
        const label = size.label
            ? `${size.width}x${size.height} - ${size.label}`
            : `${size.width}x${size.height}`;
        const scenario = scenariosFor<T>(chart, component)
            .withGroupNames(...groupNames)
            .withVisualTestConfig({
                groupUnder: groupLabel,
                screenshotSize: { width: size.width, height: size.height },
            })
            .withDefaultTags("vis-config-only", "mock-no-scenario-meta");

        if (customizer) {
            scenario.addScenarios(label, baseProps, customizer);
        } else {
            scenario.addScenario(label, baseProps);
        }

        return scenario;
    });
}
