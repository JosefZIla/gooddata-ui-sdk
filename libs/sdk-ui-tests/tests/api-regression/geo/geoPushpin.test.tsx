// (C) 2007-2019 GoodData Corporation

import geoScenarios, { latitudeLongitudeScenarios } from "../../../scenarios/geo/index";
import { ScenarioAndDescription } from "../../../src";
import { mountChartAndCapture } from "../../_infra/render";
import { IGeoPushpinChartProps, IGeoPushpinChartLatitudeLongitudeProps } from "@gooddata/sdk-ui-geo";
import flatMap from "lodash/flatMap";
import { createInsightDefinitionForChart } from "../../_infra/insightFactory";
import { cleanupGeoChartProps } from "../../_infra/utils";
import { mountInsight } from "../../_infra/renderPlugVis";
import { defSetSorts } from "@gooddata/sdk-model";
import { ReactWrapper } from "enzyme";

const Chart = "GeoPushpinChart";

describe(Chart, () => {
    const Scenarios: Array<ScenarioAndDescription<IGeoPushpinChartProps>> = flatMap(geoScenarios, (group) =>
        group.forTestTypes("api").asScenarioDescAndScenario(),
    );

    const latLongScenarios: Array<ScenarioAndDescription<IGeoPushpinChartLatitudeLongitudeProps>> = flatMap(
        latitudeLongitudeScenarios,
        (group) => group.forTestTypes("api").asScenarioDescAndScenario(),
    );

    describe.each(Scenarios)("with %s", (_desc, scenario) => {
        const promisedInteractions = mountChartAndCapture(scenario, (wrapper: ReactWrapper) => {
            return wrapper.find("GeoChartOptionsWrapper").props();
        });

        it("should create expected execution definition", async () => {
            const interactions = await promisedInteractions;

            expect(interactions.triggeredExecution).toMatchSnapshot();
        });

        it("should create expected props for core chart", async () => {
            const interactions = await promisedInteractions;

            expect(interactions.effectiveProps).toBeDefined();
            expect(interactions.effectiveProps!.execution).toBeDefined();
            expect(cleanupGeoChartProps(interactions.effectiveProps)).toMatchSnapshot();
        });

        it("should lead to same execution when rendered as insight via plug viz", async () => {
            const interactions = await promisedInteractions;

            const insight = createInsightDefinitionForChart(Chart, _desc, interactions);
            const plugVizInteractions = await mountInsight(scenario, insight);

            // remove sorts from both original and plug viz exec - simply because plug vis will automatically
            // create sorts
            const originalExecutionWithoutSorts = defSetSorts(interactions.triggeredExecution!);
            const executionWithoutSorts = defSetSorts(plugVizInteractions.triggeredExecution!);

            expect(executionWithoutSorts).toEqual(originalExecutionWithoutSorts);
        });
    });

    describe.each(latLongScenarios)("with %s", (_desc, scenario) => {
        const promisedInteractions = mountChartAndCapture(scenario, (wrapper: ReactWrapper) => {
            return wrapper.find("GeoChartOptionsWrapper").props();
        });

        it("should create expected execution definition", async () => {
            const interactions = await promisedInteractions;

            expect(interactions.triggeredExecution).toMatchSnapshot();
        });

        it("should create expected props for core chart", async () => {
            const interactions = await promisedInteractions;

            expect(interactions.effectiveProps).toBeDefined();
            expect(interactions.effectiveProps!.execution).toBeDefined();
            expect(cleanupGeoChartProps(interactions.effectiveProps)).toMatchSnapshot();
        });

        it("should lead to same execution when rendered as insight via plug viz", async () => {
            const interactions = await promisedInteractions;

            const insight = createInsightDefinitionForChart(Chart, _desc, interactions);
            const plugVizInteractions = await mountInsight(scenario, insight);

            // remove sorts from both original and plug viz exec - simply because plug vis will automatically
            // create sorts
            const originalExecutionWithoutSorts = defSetSorts(interactions.triggeredExecution!);
            const executionWithoutSorts = defSetSorts(plugVizInteractions.triggeredExecution!);

            expect(executionWithoutSorts).toEqual(originalExecutionWithoutSorts);
        });
    });
});
