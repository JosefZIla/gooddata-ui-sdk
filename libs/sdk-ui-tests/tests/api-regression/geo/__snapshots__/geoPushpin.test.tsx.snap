// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeoPushpinChart with latitude and longitude only with tooltip should create expected execution definition 1`] = `
Object {
  "attributes": Array [
    Object {
      "attribute": Object {
        "displayForm": Object {
          "identifier": "label.uscities.city.location",
          "type": "displayForm",
        },
        "localIdentifier": "a_label.uscities.city.location",
      },
    },
    Object {
      "attribute": Object {
        "displayForm": Object {
          "identifier": "label.uscities.city.location",
          "type": "displayForm",
        },
        "localIdentifier": "longitude_df",
      },
    },
    Object {
      "attribute": Object {
        "displayForm": Object {
          "identifier": "label.uscities.city",
          "type": "displayForm",
        },
        "localIdentifier": "tooltipText_df",
      },
    },
  ],
  "buckets": Array [
    Object {
      "items": Array [],
      "localIdentifier": "size",
    },
    Object {
      "items": Array [],
      "localIdentifier": "color",
    },
    Object {
      "items": Array [
        Object {
          "attribute": Object {
            "displayForm": Object {
              "identifier": "label.uscities.city.location",
              "type": "displayForm",
            },
            "localIdentifier": "a_label.uscities.city.location",
          },
        },
      ],
      "localIdentifier": "latitude",
    },
    Object {
      "items": Array [
        Object {
          "attribute": Object {
            "displayForm": Object {
              "identifier": "label.uscities.city.location",
              "type": "displayForm",
            },
            "localIdentifier": "longitude_df",
          },
        },
      ],
      "localIdentifier": "longitude",
    },
    Object {
      "items": Array [],
      "localIdentifier": "segment",
    },
    Object {
      "items": Array [
        Object {
          "attribute": Object {
            "displayForm": Object {
              "identifier": "label.uscities.city",
              "type": "displayForm",
            },
            "localIdentifier": "tooltipText_df",
          },
        },
      ],
      "localIdentifier": "tooltipText",
    },
  ],
  "dimensions": Array [
    Object {
      "itemIdentifiers": Array [
        "a_label.uscities.city.location",
        "longitude_df",
        "tooltipText_df",
      ],
    },
  ],
  "filters": Array [],
  "measures": Array [],
  "postProcessing": Object {},
  "sortBy": Array [],
  "workspace": "live-examples-workspace",
}
`;

exports[`GeoPushpinChart with latitude and longitude only with tooltip should create expected props for core chart 1`] = `
Object {
  "config": Object {
    "mapboxToken": "this-is-not-real-token",
    "showLabels": false,
    "tooltipText": Object {
      "attribute": Object {
        "displayForm": Object {
          "identifier": "label.uscities.city",
          "type": "displayForm",
        },
        "localIdentifier": "tooltipText_df",
      },
    },
  },
  "error": null,
  "exportTitle": "GeoPushpinChart",
  "isLoading": true,
  "onDataTooLarge": [Function],
  "onNegativeValues": [Function],
  "theme": Object {},
  "themeIsLoading": undefined,
}
`;

exports[`GeoPushpinChart with location only should create expected execution definition 1`] = `
Object {
  "attributes": Array [
    Object {
      "attribute": Object {
        "displayForm": Object {
          "identifier": "label.uscities.city.location",
          "type": "displayForm",
        },
        "localIdentifier": "a_label.uscities.city.location",
      },
    },
  ],
  "buckets": Array [
    Object {
      "items": Array [],
      "localIdentifier": "size",
    },
    Object {
      "items": Array [],
      "localIdentifier": "color",
    },
    Object {
      "items": Array [
        Object {
          "attribute": Object {
            "displayForm": Object {
              "identifier": "label.uscities.city.location",
              "type": "displayForm",
            },
            "localIdentifier": "a_label.uscities.city.location",
          },
        },
      ],
      "localIdentifier": "location",
    },
    Object {
      "items": Array [],
      "localIdentifier": "segment",
    },
  ],
  "dimensions": Array [
    Object {
      "itemIdentifiers": Array [
        "a_label.uscities.city.location",
      ],
    },
  ],
  "filters": Array [],
  "measures": Array [],
  "postProcessing": Object {},
  "sortBy": Array [],
  "workspace": "live-examples-workspace",
}
`;

exports[`GeoPushpinChart with location only should create expected props for core chart 1`] = `
Object {
  "config": Object {
    "mapboxToken": "this-is-not-real-token",
    "showLabels": false,
  },
  "error": null,
  "exportTitle": "GeoPushpinChart",
  "isLoading": true,
  "onDataTooLarge": [Function],
  "onNegativeValues": [Function],
  "theme": Object {},
  "themeIsLoading": undefined,
}
`;
